.github/workflows/deploy-public-build.yml

```diff
@@ -0,0 +1,37 @@
+# Adapted from https://github.com/marketplace/actions/deploy-pr-preview.
+name: Deploy Public Build
+
+on:
+  push:
+    branches:
+      - main
```

Comment:
one thing to consider is doing versioned releases where we only do this build based on whether the push to main has a tag starting with `v`.  Maybe a separate PR since it adds a bit more infrastructure.